### ISSUE TITLE
createType(StorageKey):: Derived TypedArray constructor created an array which was too small

### ISSUE THREAD

**Issue by xwartz** [polkadot-js/api#2161]
Unable to creates a unsigned extrinsic:
```ts
import Decorated from '@polkadot/metadata/Decorated'
import { TypeRegistry } from '@polkadot/types/create'
import ExtrinsicV4 from '@polkadot/types/extrinsic/v4/Extrinsic'
import metadataStatic from '@polkadot/metadata/Metadata/static'

const registry = new TypeRegistry()
const decorated = new Decorated(registry, metadataStatic)

new ExtrinsicV4(
    registry,
    decorated.tx.balances.transfer('5HpLdCTNBQDjFomqpG2XWadgB4zHTuqQqNHhUyYbett7k1RR', 0),
  ).toHex()
```

error message:

```
createType(StorageKey):: Derived TypedArray constructor created an array which was too small
```

version:

```
"@polkadot/metadata": "1.10.1",
"@polkadot/types": "1.10.1",
```

**Comment by jacogr** [comment]
I honestly have not seen that syntax uses anywhere. So what are you trying to achieve?

**Comment by xwartz** [comment]
@jacogr   I want to creates a unsigned extrinsic, from the [Extrinsic.spec.ts](https://github.com/polkadot-js/api/blob/master/packages/types/src/extrinsic/v4/Extrinsic.spec.ts#L32)

**Comment by jacogr** [comment]
Unsigned is simply -

```js
const tx = api.tx.balances.transfer('5HpLdCTNBQDjFomqpG2XWadgB4zHTuqQqNHhUyYbett7k1RR', 0);
```

Since there has been no `.sign` or `signAndSend` called on it, it is without signature and unsigned. If you call `tx.send(...)` it will therefore send without signature.

Th reason for the "interrogation" is just because you really normally don't want to access the classes directly with a new).

**Comment by xwartz** [comment]
I found that it is not working in react-native, but it works in nodejs env, so I don't think `api.tx.balances.transfer` will solve the problem.

**Comment by jacogr** [comment]
What does not work in RN? There are no internal differences between `signAndSend` and calling `sign` (optional) and `send` (must be done) internally. If there is an issue which is specific to that environment, it should be addresses.

**Comment by xwartz** [comment]
The `new Decorated` func throws an error `createType(StorageKey):: Derived TypedArray constructor created an array which was too small`  on RN 0.59.10
```
const decorated = new Decorated(registry, metadataStatic)
```

**Comment by jacogr** [comment]
Ok, so you really don't want to use those directly, as explained above. (It is used inside tests because we are setting up limited infrastructure to test only the different areas - additionally it uses the static metadata there since these keep changing, so it is a snapshot of a known state so we can get known outputs). The API will -

- retrieve the chain metadata
- create the object
- decorate all the endpoints

(the API itself also does not use the Decorated object directly, since it really is metadata-specific in use, rather using the versioned variant)

**Comment by xwartz** [comment]
For some reason, I cannot use the api method because it only supports wsProvider, and I need to use jsonrpc provider

**Comment by jacogr** [comment]
`HttpProvider` should work (although there may be some gaps), however it does not support subscriptions. So for instance, you won't be able to -

- query storage via subscriptions
- use `.sign` or `.signAndSend` via callback (only top-level await for hash)

If there are any construction issues, they must be addressed.

**Comment by xwartz** [comment]
fixed on RN 0.59.10

1. rm -rf node_modules && yarn install
2. NODE_OPTIONS=--max_old_space_size=8192 npm start
